straxen/plugins/x1t_cuts.py

```diff
@@ -236,11 +236,16 @@ class SR1Cuts(strax.MergeOnlyPlugin):
 class FiducialEvents(strax.Plugin):
     depends_on = ['event_info', 'cut_fiducial_cylinder_1t']
     data_kind = 'fiducial_events'
+    __version__ = '0.0.1'
 
     def infer_dtype(self):
         dtype = [self.deps[d].dtype_for(d) for d in self.depends_on]
-        dtype.sort()
-        return strax.merged_dtype(dtype)
+        dtype = strax.merged_dtype(dtype)
+        return dtype
 
     def compute(self, events):
-        return events[events['cut_fiducial_cylinder_1t']]
+        fiducial_events = events[events['cut_fiducial_cylinder_1t']]
+        result = np.zeros(len(fiducial_events), dtype=self.dtype)
+        # Cast the fiducual events dtype into the expected format
+        strax.copy_to_buffer(fiducial_events, result, '_fiducial_copy')
+        return result
```

Comment:
Somehow the order of the lists of the dtype got interchanged. I read the release notes of py3.10 again but wasn't able to figure out which change could case it.

I did not look much further, as this is only a 1T plugin that no-one uses anymore and this was an easy fix.

Reply:
you can back-track my steps in the commits here and find the related error:
https://github.com/XENONnT/straxen/runs/4773432924?check_suite_focus=true

Reply:
oh ok, yeah thats really weird, i wonder what that's all about. if you ever figure it out let me know :)

Reply:
@jmosbacher I think it was the numpy change, rather than the python change which caused this. We also just ran into this issue in the base env:
https://github.com/XENONnT/base_environment/runs/4815468752?check_suite_focus=true

Reply:
Probably has to do with the changes made to the dtype in 1.21:
https://numpy.org/doc/stable/release/1.21.0-notes.html#changes-to-comparisons-with-dtype